ctru-rs/src/lib.rs

```diff
@@ -15,10 +15,50 @@ extern crate pthread_3ds;
 #[cfg(feature = "big-stack")]
 static __stacksize__: usize = 2 * 1024 * 1024; // 2MB
 
-/// Activate ´ctru-rs´' default panic handler.
+macro_rules! from_type_to_u8 {
+    ($from_type:ty) => {
+        impl From<$from_type> for u8 {
+            fn from(v: $from_type) -> Self {
+                v as u8
+            }
+        }
+    };
+}
+
+macro_rules! from_type_to_u16 {
+    ($from_type:ty) => {
+        impl From<$from_type> for u16 {
+            fn from(v: $from_type) -> Self {
+                v as u16
+            }
+        }
+    };
+}
+
+macro_rules! from_type_to_u32 {
+    ($from_type:ty) => {
+        impl From<$from_type> for u32 {
+            fn from(v: $from_type) -> Self {
+                v as u32
+            }
+        }
+    };
+}
+
+macro_rules! from_type_to_i32 {
+    ($from_type:ty) => {
+        impl From<$from_type> for i32 {
+            fn from(v: $from_type) -> Self {
+                v as i32
+            }
+        }
+    };
+}
```

Comment:
these could probably go into a `mod macros` or something. Also might be simpler to have a single macro, `from_impl!(Ty, i32` and just call it like
```rs
from_impl!(Kind, u32);
```

I'm sure there's some upstream crate that can do this for us too but seems like overkill for this simple case.